### PR TITLE
sonarqube build-wrapper 4.1

### PR DESCRIPTION
**Pull request by sv99**
build-wrapper for sonarqube C++ plugin

### CODE REVIEW

File: Formula/sonar-build-wrapper.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  homepage "http://docs.sonarqube.org/pages/viewpage.action?pageId=3080359"
+  url "https://sonarqube.com/static/cpp/build-wrapper-macosx-x86.zip"
+  version "4.1"
+  sha256 "5b6dc15d2e85edf5cd24e932e390e94cd1a6e472a07d2bef4dc9c04e696d0348"
+
+  bottle :unneeded
+
+  def install
+    bin.install "build-wrapper-macosx-x86"
+    mv bin/"build-wrapper-macosx-x86", bin/"build-wrapper"
```
Comment (DomT4) [inline Formula/sonar-build-wrapper.rb]: Just:
```ruby
bin.install "build-wrapper-macosx-x86" => "build-wrapper"
```
That's a pretty generic name though. Maybe `"sonarqube-build-wrapper"` to match the formula or something?

File: Formula/sonar-build-wrapper.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    lib.install "libinterceptor.dylib"
+  end
+
+  test do
+    assert_match /version 4\.1/, shell_output("#{bin}/build-wrapper -h", 10)
```
Comment (DomT4) [inline Formula/sonar-build-wrapper.rb]:
```ruby
assert_match version.to_s, shell_output("#{bin}/build-wrapper -h", 10)
```

**Review by DomT4 — CHANGES_REQUESTED** [review]

File: Formula/sonar-build-wrapper.rb (hunk 1 above)
Comment (sv99) [inline Formula/sonar-build-wrapper.rb]: This is name from sonarqube distro.

**Comment by sv99** [comment]
updated
May be renaming to the Formula/sonar-build-wrapper.rb

script name used in the sonarcube documentation

This formula with sonar-scanner.rb allow checking project without installing sonar server.
I check my C++ github project, local C++ sonar server need C++ plugin with license. Open source projects on the github sonarcube checked for free.

File: Formula/sonar-build-wrapper.rb (hunk 1 above)
Comment (dunn) [inline Formula/sonar-build-wrapper.rb]: It's still too generic for us; we want to avoid naming conflicts with other packages.

**Comment by sv99** [comment]
updated
leave original script name  "build-wrapper-macosx-x86"

File: Formula/sonar-build-wrapper.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  homepage "http://docs.sonarqube.org/pages/viewpage.action?pageId=3080359"
+  url "https://sonarqube.com/static/cpp/build-wrapper-macosx-x86.zip"
+  version "4.1"
+  sha256 "5b6dc15d2e85edf5cd24e932e390e94cd1a6e472a07d2bef4dc9c04e696d0348"
+
+  bottle :unneeded
+
+  def install
+    bin.install "build-wrapper-macosx-x86"
+    lib.install "libinterceptor.dylib"
```
Comment (MikeMcQuaid) [inline Formula/sonar-build-wrapper.rb]: We need to build binaries and libraries from source.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/sonar-build-wrapper.rb (hunk 3 above)
Comment (sv99) [inline Formula/sonar-build-wrapper.rb]: this script build-wrapper for C/C++ sonarqube plugin - must be downloaded from installed sonarcube server. C/C++ plugin not free, for local server need license.
For open source projects on the ginhub sonarqube.org tested for free. This formula not needed local installed sonarcube server!!

Comment (MikeMcQuaid) [inline Formula/sonar-build-wrapper.rb]: So this is not open-source? If so, we cannot include it in Homebrew.

Comment (sv99) [inline Formula/sonar-build-wrapper.rb]: sonar formula??

Comment (MikeMcQuaid) [inline Formula/sonar-build-wrapper.rb]: https://github.com/SonarSource/sonarqube ?

Comment (sv99) [inline Formula/sonar-build-wrapper.rb]: this is core open source version, many languages supported by plugins enabled and supported only in the enterprise version.

Comment (sv99) [inline Formula/sonar-build-wrapper.rb]: formula sonar-runner (i send PR for renaming to the sonar-scanner) - downloaded (not compiled)

Comment (dunn) [inline Formula/sonar-build-wrapper.rb]: Mike means that we'll need it compiled; something like 
```ruby
ENV.java_cache
system "mvn", "clean", "install"
```
in the `def install` block.

Comment (MikeMcQuaid) [inline Formula/sonar-build-wrapper.rb]: Can you link to the relevant repository?

Comment (MikeMcQuaid) [inline Formula/sonar-build-wrapper.rb]: (and what @dunn said)

Comment (sv99) [inline Formula/sonar-build-wrapper.rb]: this wrapper not open source, but his may be used for checking open source project on the github.

Documentation on the sonarqube http://docs.sonarqube.org/pages/viewpage.action?pageId=3080359 - download from working server with installed C/C++ pluging and run. Without this formula need manual download and manual install wrapper.

**Comment by MikeMcQuaid** [comment]
If the wrapper is not open-source we can't include it, sorry. Homebrew Cask may be willing to do so (but may not).

**Pull request closed without merging**